base/range.jl

```diff
@@ -860,13 +860,19 @@ end
 
 median(r::Range{<:Real}) = mean(r)
 
-function in(x, r::Range)
+function in(x::Real, r::Range{<:Real})
+    n = step(r) == 0 ? 1 : round(Integer,(x-first(r))/step(r))+1
+    n >= 1 && n <= length(r) && r[n] == x
+end
+# This method needs to be defined separately since only -(::T, ::T) needs to be supported
+# to create a range, but not necessarily -(::T, ::Real)
+function in(x::T, r::Range{<:T}) where {T}
```

Comment:
It turns out that method generates ambiguities with other methods defined in this file if I change `<:T` to `T` (as it should be). So we need to decide what choice is the most painful: requiring people to define this method themselves when they implement a custom non-`Real` type supporting ranges; or define this generic fallback here (current state of the PR), but require people to fix ambiguities when they need to define a custom method like this one.